api/daemon_test.go

```diff
@@ -24,6 +25,32 @@ func TestVersion(t *testing.T) {
 	}
 }
 
+// TestUpdate checks that /daemon/update correctly asserts that an update is
+// not available for the daemon (since the test build is always up to date).
+func TestUpdate(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	st, err := createServerTester("TestUpdate")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st.server.Close()
+
+	var update UpdateInfo
+	if err = st.getAPI("/daemon/update", &update); err != nil {
+		// Notify tester that the API call failed, but allow testing to continue.
+		// Otherwise you have to be online to run tests.
+		if strings.HasSuffix(err.Error(), errEmptyUpdateResponse.Error()) {
+			t.Skip(err)
+		}
+		t.Fatal(err)
+	}
+	if update.Available && build.Version == update.Version {
+		t.Fatal("daemon should not have an update available")
```

Comment:
seems like `update.Available` should be checked separately from the version strings.